shell/mixins/browser-tab-visibility.js

```diff
@@ -0,0 +1,24 @@
+import { NORMAN } from '@shell/config/types';
+
+export default {
+  methods: {
+    setTabVisibilityListener(isAdd) {
+      const method = isAdd ? 'addEventListener' : 'removeEventListener';
+
+      document[method]('visibilitychange', this.visibilityChange, true);
+    },
+
+    async visibilityChange() {
+      if (!document.hidden) {
+        await this.$store.dispatch('rancher/findAll', {
```

Comment:
Taking the plugin case further it would be cool to also have this configurable such that other plugins could provide an asnyc func. We'd need to chat about the best way to do it though

Reply:
I was able to isolate the logic, without any code needed on the layouts (thanks for that idea/feedback) other than importing the mixin.

Since this is pretty isolated (and shouldn't run for single products), passing a param to the mixin seems a bit too much (async function) but it's doable I guess....

**Mixin**
```
 function someMixin (parameter) {
   // execute something
  return {
    data: function () {
      return {
       test: true
      }
    }
  }
 }
```

**Component:**
```
 export default {
  mixins: [
   someMixin(someVar)
  ]
 }
```

Let me know if you want to move this idea forward 🙏